src/cascadia/TerminalControl/XamlUiaTextRange.cpp

```diff
@@ -89,12 +90,58 @@ namespace winrt::Microsoft::Terminal::Control::implementation
 
     winrt::Windows::Foundation::IInspectable XamlUiaTextRange::GetAttributeValue(int32_t textAttributeId) const
     {
-        // Copied functionality from Types::UiaTextRange.cpp
-        if (textAttributeId == UIA_IsReadOnlyAttributeId)
+        // Call the function off of the underlying UiaTextRange.
+        VARIANT result;
+        THROW_IF_FAILED(_uiaProvider->GetAttributeValue(textAttributeId, &result));
+
+        // Convert the resulting VARIANT into a format that is consumable by XAML.
+        switch (result.vt)
+        {
+        case VT_BSTR:
+        {
+            return box_value(result.bstrVal);
+        }
+        case VT_I4:
+        {
+            // Surprisingly, `long` is _not_ a WinRT type.
+            // So we have to use `int32_t` to make sure this is output properly.
+            // Otherwise, you'll get "Attribute does not exist" out the other end.
+            return box_value<int32_t>(result.lVal);
+        }
+        case VT_R8:
+        {
+            return box_value(result.dblVal);
+        }
+        case VT_BOOL:
         {
-            return winrt::box_value(false);
+            return box_value(result.boolVal);
         }
-        else
+        case VT_UNKNOWN:
+        {
+            // This one is particularly special.
+            // We might return a special value like UiaGetReservedMixedAttributeValue
+            //  or UiaGetReservedNotSupportedValue.
+            // Some text attributes may return a real value, however, none of those
+            //  are supported at this time.
+            // So we need to figure out what was actually intended to be returned.
+
+            // use C++11 magic statics to make sure we only do this once.
+            static const auto mixedAttributeVal = []() {
+                IUnknown* resultRaw;
+                com_ptr<IUnknown> result;
+                UiaGetReservedMixedAttributeValue(&resultRaw);
+                result.attach(resultRaw);
+                return result;
+            }();
+
+            if (result.punkVal == mixedAttributeVal.get())
+            {
+                return Windows::UI::Xaml::DependencyProperty::UnsetValue();
+            }
+
+            __fallthrough;

```

Comment:
Isn't it `[[fallthrough]]` nowadays?

Reply:
```
6>D:\Terminal\src\types\UiaTextRangeBase.cpp(480,23): error C2220: the following warning is treated as an error
6>D:\Terminal\src\types\UiaTextRangeBase.cpp(480,23): warning C4468: 'fallthrough': attribute must be followed by a case label or a default label
6>D:\Terminal\src\types\UiaTextRangeBase.cpp(480,23): message : to simplify migration, consider the temporary use of /Wv:18 flag with the version of the compiler with which you used to build without warnings
6>D:\Terminal\src\types\UiaTextRangeBase.cpp(741,23): warning C4468: 'fallthrough': attribute must be followed by a case label or a default label
```

_Apparently_, that's not allowed over in UiaTextRangeBase though :(